### PR TITLE
Set optimization level from command line

### PR DESCRIPTION
**Pull request by Hecate2**
It seems command line users cannot trigger --no-optimize or the experimental optimizers🤣, although they can be set in tests.
To test this PR, use the following flags:
` --debug --ex-optimize --assembly`
` --debug --no-optimize --assembly`
` --debug --no-optimize --ex-optimize --assembly` (should fail)
` --debug --ex-optimize --no-optimize --assembly` (should fail)
and read `bin/sc/{contract}.nef.txt` to see whether the contract has been correctly optimized.

### CODE REVIEW

File: src/Neo.Compiler.CSharp/CompilationOptions.cs [hunk 1]
```diff
@@ -33,6 +33,30 @@ public enum OptimizationType : byte
         public OptimizationType Optimize { get; set; } = OptimizationType.Basic;
         public bool Checked { get; set; }
         public bool NoInline { get; set; }
+        protected bool _NoOptimize;
+        public bool NoOptimize
+        {
+            get => _NoOptimize; set
+            {
+                _NoOptimize = value;
```
Comment (shargon) [inline src/Neo.Compiler.CSharp/CompilationOptions.cs:41]:
```suggestion
        public bool NoOptimize
        {
            get => Optimize == OptimizationType.None; set
            {
```

**Review by Jim8y — DISMISSED** [review]

**Comment by shargon** [comment]
> It seems command line users cannot trigger --no-optimize 

What about use --optimize=none ?

**Comment by Hecate2** [comment]
> > It seems command line users cannot trigger --no-optimize 
> 
> What about use --optimize=none ? 

That's great for me

**Comment by shargon** [comment]
I think that we should instruct the user to use --optimize=Level

**Comment by Hecate2** [comment]
` --debug --optimize=All --assembly`
` --debug --optimize=Basic --assembly`
` --debug --optimize=None --assembly`

**Review by shargon — APPROVED** [review]

**Review by Jim8y — APPROVED** [review]

**Pull request merged**